torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -121,9 +121,12 @@ c10::intrusive_ptr<JitFuture> wrapFutureMessageInJitFuture(
   if (hasValue) {
     c10::intrusive_ptr<JitFuture> jitFuture =
         c10::make_intrusive<JitFuture>(PyObjectType::get());
-
+    // Save and pass thread local state into the callback
```

Comment:
Let's add a comment to explain that this should light when profiler is disabled.

Reply:
Added